redis/src/main/scala/zio/redis/RedisCommand.scala

```diff
@@ -5,8 +5,12 @@ import zio.ZIO
 final class RedisCommand[-In, +Out] private (val name: String, val input: Input[In], val output: Output[Out]) {
   private[redis] def run(in: In): ZIO[RedisExecutor, RedisError, Out] =
     ZIO
-      .accessM[RedisExecutor](_.get.execute(Input.StringInput.encode(name) ++ input.encode(in)))
-      .flatMap(out => ZIO.effect(output.unsafeDecode(out)))
+      .accessM[RedisExecutor] { executor =>
+        val service = executor.get
+        val codec   = service.codec
+        val command = Input.StringInput.encode(name)(codec) ++ input.encode(in)(codec)
+        service.execute(command).flatMap[Any, Throwable, Out](out => ZIO.effect(output.unsafeDecode(out)(codec)))
```

Comment:
Interesting thing without explicit types in flatMap scala 2.12.13 doesn't compile with error `a type was inferred to be `Any`; this may indicate a programming error.`

Reply:
Yes, this is a very annoying issue.